capi/geos_c.h.in

```diff
@@ -3635,6 +3658,69 @@ extern GEOSGeometry GEOS_DLL *GEOSOffsetCurve(const GEOSGeometry* g,
 
 ///@}
 
+
+/* ====================================================================== */
+/** @name Coverages
+* Functions to work with coverages represented by lists of polygons
+* that exactly share edge geometry.
+*/
+///@{
+
+/**
+* Analyze a coverage (represented as a collection of polygonal geometry
+* with exactly matching edge geometry) to find places where the
+* assumption of exactly matching edges is not met.
+*
+* \param input The polygonal coverage to access,
+*        stored in a geometry collection.
+* \return A value of 1 for a valid coverage, 0 for invalid and -1 for
+*         a processing error.
+*
+* \since 3.12
+*/
+extern int GEOSCoverageIsValid(
+    const GEOSGeometry* input);
+
+/**
+* Analyze a coverage (represented as a collection of polygonal geometry
+* with exactly matching edge geometry) to find places where the
+* assumption of exactly matching edges is not met.
+*
+* \param input The polygonal coverage to access,
+*        stored in a geometry collection.
+* \param gapWidth The maximum width of gaps to detect.
+* \return A collection containing a MULTILINESTRING indicating the
+*         invalid edges for each input polygon, or an empty
+*         MULTILINESTRING for polygons that are valid.
+*
+* \since 3.12
+*/
+extern GEOSGeometry GEOS_DLL *GEOSCoverageValid(
```

Comment:
Perhaps `GEOSCoverageInvalidEdges` would be more descriptive of the expected output, which are instances of invalid edges?

What will this return if there is a processing error (e.g., inputs have things other than polygons, etc)?  Perhaps `NULL` signals a failure?

Reply:
> What will this return if there is a processing error (e.g., inputs have things other than polygons, etc)? Perhaps `NULL` signals a failure?

Other GEOS functions (e.g. [`GEOSDelaunayTriangulation`](http://libgeos.org/doxygen/geos__c_8h.html#a4838d3740f3919fb4a97b349eef94278) ) return NULL to indicate some kind of error.  So makes sense to follow that pattern here.